alfasim_sdk/types.py

```diff
@@ -113,11 +113,28 @@ def check(self, attr: Attribute, values: List[str]) -> None:  # pylint: disable=
 @attr.s(kw_only=True)
 class Reference(BaseField):
     ref_type = attrib()
+    container_type = attrib(default=None, validator=[optional(instance_of(str)), optional(check_string_is_not_empty)])
+
+    def __attrs_post_init__(self):
+        if issubclass(self.ref_type, ALFAsimType):
+            self.container_type = attr.fields(self.ref_type).container_type.default
```

Comment:
```suggestion
            self.container_type = self.ref_type._CONTAINER_TYPE
```

Following my comment about this being a class attribute.